app/app/src/main/java/com/example/clubwat/viewmodels/ClubDiscussionViewModel.kt

```diff
@@ -0,0 +1,125 @@
+package com.example.clubwat.viewmodels
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.clubwat.model.ClubDetails
+import com.example.clubwat.model.Data
+import com.example.clubwat.model.NetworkResult
+import com.example.clubwat.model.ProcessedData
+import com.example.clubwat.model.SendDiscusionMessageRequest
+import com.example.clubwat.repository.DiscussionRepository
+import com.example.clubwat.repository.UserRepository
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+class ClubDiscussionViewModel(
+    val userRepository: UserRepository,
+    private val discussionRepository: DiscussionRepository
+) : ViewModel() {
+    private val _uiState = MutableStateFlow(DiscussionUiState.initial)
+    val uiState: StateFlow<DiscussionUiState> = _uiState
+
+    fun fetchUpdatedPosts(clubId: String) {
+        viewModelScope.launch(Dispatchers.IO) {
+            when (val response = discussionRepository.getPosts(
+                clubId,
+                userRepository.currentUser.value?.userId.toString()
+            )) {
+                is NetworkResult.Success -> {
+                    _uiState.emit(
+                        _uiState.value.copy(
+                            posts = processData(response.data?.data ?: emptyList())
+                        )
+                    )
+                    startPolling(clubId)
+                }
+
+                is NetworkResult.Error -> {
+                    // Can be user to handle errors in future...
+                    delay(500)
+                    fetchUpdatedPosts(clubId) // try again after 5 seconds
+                }
+            }
+        }
+    }
+
+    private fun startPolling(clubId: String) {
+        viewModelScope.launch {
+            while (true) { // ONLY IN VM SCOPE
+                delay(60000) // delay 60s then refetch
```

Comment:
Is it possible to make the delay less? Why was 60s chosen?

Reply:
polling too often causes too many connections to the db stops other queries from working